core-tests/src/test/java/org/mule/runtime/core/internal/source/polling/DefaultSchedulerMessageSourceTestCase.java

```diff
@@ -75,48 +89,51 @@ public void setUp() throws Exception {
 
   @Test
   public void simplePoll() throws Exception {
-
-    DefaultSchedulerMessageSource schedulerMessageSource = createMessageSource();
-
-    SensingNullMessageProcessor flow = getSensingNullMessageProcessor();
-    schedulerMessageSource.setListener(flow);
-    schedulerMessageSource.setAnnotations(singletonMap(LOCATION_KEY, TEST_CONNECTOR_LOCATION));
-
-    doAnswer(invocationOnMock -> {
-      CoreEvent inputEvent = invocationOnMock.getArgument(0);
-      flow.process(inputEvent);
-      return null;
-    }).when(sourcePolicy).process(any(CoreEvent.class), any(), any());
-
-    schedulerMessageSource.trigger();
-    new PollingProber(RECEIVE_TIMEOUT, 100).check(new Probe() {
+    pollAndAssertWith(p -> new PollingProber(RECEIVE_TIMEOUT, 100).check(new Probe() {
 
       @Override
       public boolean isSatisfied() {
-        return flow.event != null;
+        return p.event != null;
       }
 
       @Override
       public String describeFailure() {
         return "flow event never set by the source flow";
       }
-    });
+    }));
   }
 
   @Test
-  public void disposeScheduler() throws Exception {
-    SchedulerService schedulerService = muleContext.getSchedulerService();
-    reset(schedulerService);
+  @Description("Verifies that no events are processed when the scheduler is disabled")
+  @Issue("MULE-18063")
+  public void simplePollDisabledWithScheduler() throws Exception {
+    String wasDisabled = setProperty(MULE_DISABLE_SCHEDULERS, "true");
```

Comment:
make this test parameterized and use a rule for the system property

Reply:
Done.  I split this TestCase in two to make one of them parameterized.